### PR TITLE
CHANGELOG: Add changelog for 2.0.5

### PR DESCRIPTION
**Pull request by shenli**
Thank you for working on TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.

## What have you changed? (mandatory)
CHANGELOG.md

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
Add changelog for 2.0.5

## What are the type of the changes (mandatory)?
Improvement

## How has this PR been tested (mandatory)?
No need to test

Please describe the tests that you ran to verify your changes. Have you finished unit tests, integration tests, or manual tests?

## Does this PR affect documentation (docs/docs-cn) update? (optional)
Yes, we will update the release pages in docs/docs-cn

## Refer to a related PR or issue link (optional)
None

## Benchmark result if necessary (optional)
None

## Add a few positive/negative examples (optional)
None

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,28 @@
...
 
+## [2.0.5] - 2018-07-06
+### New Features
+  - Add the `tidb_disable_txn_auto_retry` system variable, used to disable the automatic retry of transactions #6877
+### Improvements
+  - Optimize the cost calculation of `Selection` to make the result more accurate #6989
+  - Select the query condition that completely matches the unique index or the primary key as the query path directly #6966
+  - Execute necessary cleanup when failing to start the service #6964
+  - Handle `\N` as NULL in the `Load Data` statement #6962
+  - Optimize the code structure of COB #6953
```
Comment (winoros) [inline CHANGELOG.md]: `COB` -> `CBO`

**Comment by shenli** [comment]
@winoros PTAL

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,28 @@
 
 All notable changes to this project will be documented in this file. See also [Release Notes](https://github.com/pingcap/docs/blob/master/releases/rn.md), [TiKV changelog](https://github.com/pingcap/tikv/blob/master/CHANGELOG.md) and [PD changelog](https://github.com/pingcap/pd/blob/master/CHANGELOG.md).
 
+## [2.0.5] - 2018-07-06
+### New Features
+  - Add the `tidb_disable_txn_auto_retry` system variable, used to disable the automatic retry of transactions #6877
```
Comment (winoros) [inline CHANGELOG.md]: Do we need to change the plain test `#6877` to a link  [#6867](https://github.com/pingcap/tidb/pull/6877). Or just remove it?

Comment (jackysp) [reply CHANGELOG.md]: "which is used to disable xxx"?

**Comment by jackysp** [comment]
@lilin90 PTAL

**Pull request merged**